packages/camera/camera/ios/Classes/QueueUtils.m

```diff
@@ -2,18 +2,14 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-#import "QueueHelper.h"
+#import "QueueUtils.h"
 
 const char *FLTCaptureSessionQueueSpecific = "capture_session_queue";
 
-@implementation QueueHelper
-
-+ (void)ensureToRunOnMainQueue:(void (^)(void))block {
+void FLTEnsureToRunOnMainQueue(dispatch_block_t block) {
```

Comment:
Frankly I still don't think these little utility function or classes should exist, it's just a wrapper around 3 lines of well-understood Foundation code, your Objective-C eye can just skim over some code about `NSThread` instead of needing to understand what `FLTEnsureToRunOnMainQueue` does.  In my opinion it would be clearer to just check if you're on the main thread in the places you need to.  And then you have access to `_cmd` 
https://github.com/flutter/plugins/blob/10a59af20f143d52d5a2c471bb9e7e3e79367a1e/packages/video_player/video_player_avfoundation/ios/Classes/FLTVideoPlayerPlugin.m#L349

Reply:
Some params are not objects and it's annoying to wrap/unwrap them, so I will not use `performSelector` approach.